ResearchKit/Common/ORKResultPredicate.h

```diff
@@ -196,6 +319,24 @@ ORK_CLASS_AVAILABLE
  
  @param taskIdentifier      The identifier of the task whose result you want to match. Pass `nil`
                                 to match the ongoing task.
+ @param stepIdentifier      The identifier of the step whose result you want to match. Passing `nil`
+                                assumes the step identifier is the same as the result identifier.
+ @param resultIdentifier    The identifier of the question result you are interested in.
+ @param expectedString      The expected string answer.
+ 
+ @return A result predicate.
+ */
++ (NSPredicate *)predicateForChoiceQuestionResultWithTaskIdentifier:(nullable NSString *)taskIdentifier
+                                                     stepIdentifier:(nullable NSString *)stepIdentifier
+                                                   resultIdentifier:(NSString *)resultIdentifier
+                                                     expectedString:(NSString *)expectedString;
```

Comment:
The `value` property of an `ORKTextChoice` is not limited to `NSString`.

```
@property (copy, readonly) id<NSCopying, NSCoding, NSObject> value;
```
And ORKChoiceQuestionResult only carries an array of the value object. Does this predicate require the textChoice to use a string as value object?